internal/registry/registry.go

```diff
@@ -20,64 +21,72 @@ import (
 
 func SaveImage(imagePath string, image kubernetes.ContainerImage) error {
 	imageMap := map[string]v1.Image{}
-
 	o := crane.GetOptions()
 
-	if len(image.Auth) > 0 {
-		cfg, err := ResolveAuthConfig(image)
-		empty := types.AuthConfig{}
+	for pullSecretIndex, pullSecret := range image.PullSecrets {
```

Comment:
This logic is good in general, except the fact that it assumes, that there is always at least one pull-secret. That's wrong. The pull-secrets are always optional.

Reply:
yeah - you are right
made some changes to the pullSecret iteration handling

Reply:
one commit is still missing (returning error instead of nil when necessary)

Reply:
done
the pullsecret handling should work now and i added some tests